src/Symfony/Component/HttpKernel/Bundle/SimpleBundle.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Bundle;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+/**
+ * An implementation of a simple all-in-one {@link BundleInterface} and
+ * configurable {@link Extension}.
+ *
+ * @author Guilhem N. <egetick@gmail.com>
+ */
+abstract class SimpleBundle extends Bundle implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public function getContainerExtension()
+    {
+        if (null === $this->extension) {
+            $this->extension = new SimpleExtension($this);
+        }
+
+        return $this->extension;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root($this->getContainerExtension()->getAlias());
+
+        $this->buildConfiguration($rootNode);
+
+        return $treeBuilder;
+    }
+
+    protected function buildConfiguration(TreeBuilder $rootNode)
+    {
+    }
+
+    protected function buildContainer(array $config, ContainerBuilder $container)
```

Comment:
I really dont like this name :confused: imo. it's confusing with `Bundle::build(ContainerBuilder $container)`.

I think it makes more sense to have `SimpleBundle::loadExtension` (as we're always called from `SimpleExtension::load`).

> What about configureBundle ? or configureContainer ?

We really should consider having `Bundle::build` already and what is best DX. So.., it's called from `Kernel::prepareContainer`, which is called from `Kernel::buildContainer`. Ie. it's easy to assume `build` already acts as `buildContainer`, it's just poor naming imo.

However, in that spirit, we could also go with simply `SimpleBundle::load`.